catalog/mvcc/cataloger_list_entries.go

```diff
@@ -241,6 +251,8 @@ func processSinglePrefix(response []entryPathPrefixInfo, delimiter string, branc
 }
 
 func getMoreRows(path string, branch int64, branchRanges map[int64][]entryPathPrefixInfo, readParams readParamsType) error {
+	// if the buffer initially read is exhausted (e.g. - all entries that were read were deleted) - more rows are read for that branch
+	// results are directly stored in the branchRanges map. (agreed - there are simpler and clearer ways to do it)
```

Comment:
This comment is a bit hard to understand, because it is in the passive voice.  E.g.
> getMoreRows reads more rows for a branch after all its read are exhausted from (_I don't know where, actually_)..., and stores those results directly into branchRanges.